### PR TITLE
Devdocs

### PR DESCRIPTION
**Pull request by fbs**
##### Checklist

- [ ] Language changes are updated in `docs/reference_guide.md`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -8,3 +8,4 @@ Then please complete the checklist.
 
 - [ ] Language changes are updated in `docs/reference_guide.md`
 - [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
+- [ ] The new behaviour is covered by tests
```
Comment (javierhonduco) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: 😍

**Comment by javierhonduco** [comment]
LGTM! Do we wanna ask folks to add the formatting changes in a commit below their changes?

**Comment by fbs** [comment]
> LGTM! Do we wanna ask folks to add the formatting changes in a commit below their changes?

What do you mean by 'below'? All the PRs we do should have the individual commits  correctly formatted, having a fixup commit makes git blame harder to use. I had it in the document like

> We want to avoid `fix formatting` commits. Instead every commit should be
formatted correctly.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -8,3 +8,4 @@ Then please complete the checklist.
```
Comment (fbs) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: I  think it will be good to link the document  here somehow

**Comment by javierhonduco** [comment]
What I meant is that instead of including the changes + formatting in one commit, which might make reviewing it harder, maybe doing the formatting in one commit + actual changes in another commit could be better. We can always squash them at merge time

**Comment by fbs** [comment]
> What I meant is that instead of including the changes + formatting in one commit, which might make reviewing it harder, maybe doing the formatting in one commit + actual changes in another commit could be better. We can always squash them at merge time

Hmm, I'm not a big fan of that. If we forget to squash we still end up with that 'fix formatting' commit. And on a PR that has more than one commit it won't work either.

**Comment by danobi** [comment]
> What I meant is that instead of including the changes + formatting in one commit, which might make reviewing it harder,

Hmm, `git clang-format` only formats the lines you touched. I've never had any issue with unreadability. And "formatting" commit after the "real" commits also kinda clobbers git blame.

**Pull request merged**